WordPress/Sniffs/NamingConventions/ValidFunctionNameSniff.php

```diff
@@ -142,19 +143,16 @@ protected function process_method_declaration( $stackPtr, $methodName, $currScop
 			$className = '[Anonymous Class]';
 		} else {
 			$className = ObjectDeclarations::getName( $this->phpcsFile, $currScope );
-		}
 
-		$methodNameLc = strtolower( $methodName );
-		$classNameLc  = strtolower( $className );
+			// PHP4 constructors are allowed to break our rules.
+			if ( NamingConventions::isEqual( $methodName, $className ) === true ) {
+				return;
+			}
 
-		// PHP4 constructors are allowed to break our rules.
-		if ( $methodNameLc === $classNameLc ) {
-			return;
-		}
-
-		// PHP4 destructors are allowed to break our rules.
-		if ( '_' . $classNameLc === $methodNameLc ) {
-			return;
+			// PHP4 destructors are allowed to break our rules.
+			if ( NamingConventions::isEqual( $methodName, '_' . $className ) === true ) {
+				return;
+			}
```

Comment:
I thought destructors didn't exist until PHP 5?

I've not seen them suggested with a leading underscore in PHP 4 as:

```php
class Foo {
	function Foo() {} // Constructor.
	function _Foo() {} // Destructor.
}
```
Was that the _de facto_ approach back then? How did they get triggered?

Reply:
All good questions. To be honest, it's so far back, that this should be redundant and I hope to never come across that type of code again.

For the purposes of this PR, I've just kept the existing functionality, which already handled things in this way (whether correct or not).
Also note that the PHPCS native `PEAR.NamingConventions.ValidFunctionName` sniff, which this sniff was originally based on, treats `_ClassName()` methods the same, i.e. as PHP 4-destructors.

I realize this is not an answer to your question, but there's only so much time (research) I'm willing to spend on work around for code which shouldn't exist anymore in the first place.